### ISSUE TITLE
Chalice crashes on packaging pandas

### ISSUE THREAD

**Issue by vfilimonov** [aws/chalice#1398]
Hello

After the update to 1.13.1 Chalice could not package `pandas` anymore. 

Minimal example (command line):
```
chalice new-project temp
echo "pandas==1.0.3" > temp/requirements.txt
cd temp
chalice package .
``` 
It hangs on "Creating deployment package." for several minutes and then crashes with exception:
```
Traceback (most recent call last):
  File "/Users/user/miniconda3/lib/python3.7/site-packages/chalice/cli/__init__.py", line 518, in main
    return cli(obj={})
  File "/Users/user/miniconda3/lib/python3.7/site-packages/click/core.py", line 764, in __call__
    return self.main(*args, **kwargs)
  File "/Users/user/miniconda3/lib/python3.7/site-packages/click/core.py", line 717, in main
    rv = self.invoke(ctx)
  File "/Users/user/miniconda3/lib/python3.7/site-packages/click/core.py", line 1137, in invoke
    return _process_result(sub_ctx.command.invoke(sub_ctx))
  File "/Users/user/miniconda3/lib/python3.7/site-packages/click/core.py", line 956, in invoke
    return ctx.invoke(self.callback, **ctx.params)
  File "/Users/user/miniconda3/lib/python3.7/site-packages/click/core.py", line 555, in invoke
    return callback(*args, **kwargs)
  File "/Users/user/miniconda3/lib/python3.7/site-packages/click/decorators.py", line 17, in new_func
    return f(get_current_context(), *args, **kwargs)
  File "/Users/user/miniconda3/lib/python3.7/site-packages/chalice/cli/__init__.py", line 447, in package
    packager.package_app(config, out, stage)
  File "/Users/user/miniconda3/lib/python3.7/site-packages/chalice/package.py", line 920, in package_app
    config, chalice_stage_name)
  File "/Users/user/miniconda3/lib/python3.7/site-packages/chalice/package.py", line 82, in construct_resources
    self._build_stage.execute(config, resources)
  File "/Users/user/miniconda3/lib/python3.7/site-packages/chalice/deploy/deployer.py", line 995, in execute
    step.handle(config, resource)
  File "/Users/user/miniconda3/lib/python3.7/site-packages/chalice/deploy/deployer.py", line 851, in handle
    handler(config, resource)
  File "/Users/user/miniconda3/lib/python3.7/site-packages/chalice/deploy/deployer.py", line 878, in handle_deploymentpackage
    config.project_dir, config.lambda_python_version)
  File "/Users/user/miniconda3/lib/python3.7/site-packages/chalice/deploy/packager.py", line 95, in create_deployment_package
    abi, requirements_filepath, site_packages_dir)
  File "/Users/user/miniconda3/lib/python3.7/site-packages/chalice/deploy/packager.py", line 538, in build_site_packages
    abi, tempdir, requirements_filepath)
  File "/Users/user/miniconda3/lib/python3.7/site-packages/chalice/deploy/packager.py", line 433, in _download_dependencies
    sdists = self._find_sdists(directory)
  File "/Users/user/miniconda3/lib/python3.7/site-packages/chalice/deploy/packager.py", line 349, in _find_sdists
    in self._osutils.get_directory_contents(directory)]
  File "/Users/user/miniconda3/lib/python3.7/site-packages/chalice/deploy/packager.py", line 348, in <listcomp>
    packages = [Package(directory, filename) for filename
  File "/Users/user/miniconda3/lib/python3.7/site-packages/chalice/deploy/packager.py", line 554, in __init__
    self._name, self._version = self._calculate_name_and_version()
  File "/Users/user/miniconda3/lib/python3.7/site-packages/chalice/deploy/packager.py", line 615, in _calculate_name_and_version
    sdist_path)
  File "/Users/user/miniconda3/lib/python3.7/site-packages/chalice/deploy/packager.py", line 685, in get_package_name_and_version
    pkg_info_filepath = self._generate_egg_info(package_dir)
  File "/Users/user/miniconda3/lib/python3.7/site-packages/chalice/deploy/packager.py", line 666, in _generate_egg_info
    egg_info_dir, info_contents[0], 'PKG-INFO')
IndexError: list index out of range
```

**Comment by jamesls** [comment]
Yeah, looks like a regression in 1.13.0.  There were some packaging related bug fixes that were merged, not seeing this issue in 1.12.0.  Working on this now.  Thanks for reporting.

**Comment by jamesls** [comment]
Yeah, looks like a regression in 1.13.0.  There were some packaging related bug fixes that were merged, not seeing this issue in 1.12.0.  Working on this now.  Thanks for reporting.

**Comment by jamesls** [comment]
I pushed a fix for this (#1399).  We also have an integration test now that will ensure we don't regress on this again.  I'm also working on revamping some of this packaging code to make it more robust going forward.  I'll get a release out shortly.  Thanks again for reporting.

**Comment by PeterRyder2** [comment]
Hey There. I tried to replicate the same code as vfilimonov above but I am having a similar issue. 
I add pandas to the reqiurements.txt and ran chalice deploy 

It hangs at:

Creating lambda function: temp2-dev

I was wondering is this issue resolved for trying to deploy with pandas. 

Many thanks
